### DISCUSSION TITLE
[web-bluetooth] device addEventListener advertisementreceived missing declaration/override for options parameter

### DISCUSSION THREAD

**Discussion by ariccio** [DefinitelyTyped/DefinitelyTyped#57871] [Issues with a @types package]
Currently, this function is declared as such:
```
    addEventListener(type: "advertisementreceived", listener: (this: this, ev: Event) => any, useCapture?: boolean): void;
```
...however, there should be a overload of this that accepts an `options` object, as per [MDN](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener). 

You can see an example in the web-bluetooth sample code:
https://googlechrome.github.io/samples/web-bluetooth/watch-advertisements-and-connect-async-await.html

[a comment by an automated account is not shown]

**Comment by DaBs** [comment]
Good find, I'll update the typings to properly reflect this within the week. If you have a PR for it before then, I could simply review the changes.

**Comment by thegecko** [comment]
Although not tied to specific event types, the interfaces with `addEventListener` do inherit from `EventTarget` meaning the generic `addEventListener` with options is currently available:

https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener

**Comment by ariccio** [comment]
@thegecko I see that it inherits, yes, but this line of code (with the [same syntax as a sample](https://googlechrome.github.io/samples/web-bluetooth/watch-advertisements-and-connect-async-await.html)), causes a typescript error:

```
firstDevice.addEventListener('advertisementreceived', watchAdvertisementEventReceived, {once: true});
```
Error:

```
No overload matches this call.
  Overload 1 of 3, '(type: "gattserverdisconnected", listener: (this: BluetoothDevice, ev: Event) => any, useCapture?: boolean | undefined): void', gave the following error.
    Argument of type '"advertisementreceived"' is not assignable to parameter of type '"gattserverdisconnected"'.
  Overload 2 of 3, '(type: "advertisementreceived", listener: (this: BluetoothDevice, ev: Event) => any, useCapture?: boolean | undefined): void', gave the following error.
    Argument of type '(device: BluetoothDevice, event: BluetoothAdvertisingEvent) => Promise<void>' is not assignable to parameter of type '(this: BluetoothDevice, ev: Event) => any'.
  Overload 3 of 3, '(type: string, listener: EventListenerOrEventListenerObject, useCapture?: boolean | undefined): void', gave the following error.
    Argument of type '(device: BluetoothDevice, event: BluetoothAdvertisingEvent) => Promise<void>' is not assignable to parameter of type 'EventListenerOrEventListenerObject'.
      Type '(device: BluetoothDevice, event: BluetoothAdvertisingEvent) => Promise<void>' is not assignable to type 'EventListener'.ts(2769)
```

I have to revert to my C-programmer ways:

```
(firstDevice.addEventListener as any)('advertisementreceived', watchAdvertisementEventReceived, {once: true});
```

**Reply by thegecko** [reply]
Ack

**Comment by ariccio** [comment]
Hmm, watchAdvertisements seems to be missing too.
```
watchAdvertisements(): Promise<void>;
```

I don't blame anybody, nobody is really using this yet! I think the chrome team has like maybe two people even implementing this api :)

**Reply by thegecko** [reply]
https://github.com/thegecko/webbluetooth

😉
